test/e2e/agent/agent_test.go

```diff
@@ -63,4 +64,317 @@ var _ = Describe("VM Agent behavior", func() {
+			out, err := harness.CLIWithStdin(httpRepo, "apply", "-f", "-")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(`(200 OK|201 Created)`))
+
+			// Update the device with the http invalid config.
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				logrus.Infof("current device is %s", deviceId)
+				// Create ConfigProviderSpec.
+				var configProviderSpec v1alpha1.ConfigProviderSpec
+				err := configProviderSpec.FromHttpConfigProviderSpec(httpConfigInvalidPath)
+				Expect(err).ToNot(HaveOccurred())
+
+				device.Spec.Config = &[]v1alpha1.ConfigProviderSpec{configProviderSpec}
+				logrus.Infof("updating %s with config %s", deviceId, device.Spec.Config)
+			})
+
+			// Check the http config error is detected.
+			harness.WaitForDeviceContents(deviceId, "1 invalid configuration: . Error: sending HTTP Request",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "SpecValid", "False", "Invalid")
+				}, "2m")
+
+			// The behaviour will change after EDM-418.
+			harness.WaitForDeviceContents(deviceId, "the device is updated to renderedVersion: 1",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+			Eventually(harness.GetDeviceWithStatusSummary, TIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+		})
+
+		It("should report 'Unknown' after the device vm is powered-off", func() {
+			deviceId := harness.GetEnrollmentIDFromConsole()
+
+			// Approve the enrollment and wait for the device details to be populated by the agent.
+			harness.ApproveEnrollment(deviceId, testutil.TestEnrollmentApproval())
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeEmpty())
+			logrus.Infof("The device %s was approved", deviceId)
+
+			// Wait for the device to pickup enrollment and report measurements on device status.
+			Eventually(harness.GetDeviceWithStatusSystem, TIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeNil())
+			logrus.Infof("The device %s is reporting its status", deviceId)
+
+			// Shutdown the vm.
+			err := harness.VM.Shutdown()
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Unknown")))
+		})
+	})
 })
+
+var mode = 0644
+var modePointer = &mode
+
+var inlineConfig = v1alpha1.FileSpec{
+	Content: "This system is managed by flightctl.",
+	Mode:    modePointer,
+	Path:    "/etc/motd",
+}
+
+var validInlineConfig = v1alpha1.InlineConfigProviderSpec{
+	Inline: []v1alpha1.FileSpec{inlineConfig},
+	Name:   "valid-inline-config",
+}
+
+const httpRepo = `
+apiVersion: v1alpha1
+kind: Repository
+metadata:
+  name: flightctl-demos
+spec:
+  url: https://github.com/flightctl/flightctl-demos.git
+  type: http
+`
```

Comment:
let's do the httpRepo with go structures too

Reply:
Created, thanks